python/ray/serve/tests/test_deploy.py

```diff
@@ -564,6 +564,39 @@ def make_nonblocking_calls(expected, expect_blocking=False):
     make_nonblocking_calls({"2": 2})
 
 
+def test_reconfigure_with_queries(serve_instance):
+    signal = SignalActor.remote()
+
+    @serve.deployment(max_concurrent_queries=10, num_replicas=3)
+    class A:
+        def __init__(self):
+            self.state = None
+
+        def reconfigure(self, config):
+            self.state = config
+
+        async def __call__(self):
+            await signal.wait.remote()
+            return self.state["a"]
+
+    A.options(version="1", user_config={"a": 1}).deploy()
+    handle = A.get_handle()
+    refs = []
+    for _ in range(30):
+        refs.append(handle.remote())
+
+    @ray.remote(num_cpus=0)
+    def reconfigure():
+        A.options(version="1", user_config={"a": 2}).deploy()
+
+    reconfigure_ref = reconfigure.remote()
```

Comment:
Ah maybe this is the part I wasn't understanding earlier--@hankpipi why does this need to be wrapped in a remote call?

Reply:
Can't we just replace these lines with the single line `A.options(version="1", user_config={"a": 2}).deploy()`?

Reply:
`A.options(version="1", user_config={"a": 2}).deploy()` is synchronous function, but actually it will cause deadlock, because `signal.send.remote()` is called after `deploy`, so `reconfigure` operator can't acquire write lock.

Reply:
Besides, if we call `signal.send.remote()` before reconfiguration, even the code of origin version can pass this test.

Reply:
Ah of course, that makes perfect sense.  Thanks for the patient explanation!

Reply:
Ah I just remembered we have an internal flag `.deploy(_blocking=False)`, but I think your approach is better because it only uses the public API.